lnp2p/src/bolt/bolt1.rs

```diff
@@ -145,45 +145,70 @@ impl Display for Error {
 
 #[cfg(test)]
 mod test {
+    use amplify::hex::FromHex;
     use internet2::TypedEnum;
     use lightning_encoding::LightningDecode;
 
     use super::*;
     use crate::bolt::Messages;
 
     #[test]
-    fn default_init() {
+    fn bolt1_testvec() {
         let init_msg = Messages::Init(Init {
-            global_features: none!(),
-            local_features: default!(),
+            global_features: InitFeatures::default(),
+            local_features: InitFeatures::default(),
             assets: none!(),
             unknown_tlvs: none!(),
         });
-        assert_eq!(&init_msg.serialize(), &[
-            0x00, 0x10, 0x00, 0x01, 0x00, 0x00, 0x01, 0x00
-        ]);
+        assert_eq!(
+            init_msg.serialize(),
+            Vec::<u8>::from_hex("0010000100000100").unwrap()
+        );
```

Comment:
I think the current version is more easy to read and understand. Why have you reverted that part?